MarkBind/markbind
# Make the position of top navbar fixed

This should provide an alternative solution to #973, without causing too
much trouble.

**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**

• [X] Enhancement to an existing feature






**What is the rationale for this request?**

It's a bit strange that MarkBind doesn't fix the top navigation bar, nor re-shows the navbar when scrolled down. Moreover, currently there's no "back-to-top" button, which makes both navigating across the site and searching content a bit tricky. We can fix the top navbar to fix these problems.

**What changes did you make? (Give an overview)**

Changed the position of the top navbar to be fixed and move the content below down.

Replaced the `vue-strap.min.js` file from https://github.com/MarkBind/vue-strap/pull/123

**Provide some example code that this change will affect:**

na

**Is there anything you'd like reviewers to focus on?**

na

**Testing instructions:**

To enable fixed header (or navbar), simply add `class="header-fixed"` property to the `<header>` element in `header.md`

1. On full width, check if the left and right page navigation div is sticking to the page correctly
2. On full width, check if the top navbar always shows on top of any other page content
3. See if the responsive design still works fine (i.e. the collapsed topnav).
4. Remove `class="header-fixed"` property to see if the navbar is no longer fixed.

Other things to notice:

* whether the page nav (on the right) is working properly
* whether the anchor next to the headings is working properly

**Proposed commit message: (wrap lines at 72 characters)**

Make top navbar position fixed

Top navbar is responsible for both navigating across the
site and placing the search bar. MarkBind often deals
with long pages thus hiding the top navbar and making
the user to go back to top to see the top navbar may not
be a good design choice.

In this PR I fixed the top navbar's position so that it's
always accessible to the user.
